src/certificate_provider.py

```diff
@@ -65,16 +68,21 @@ def update(self):
             if self.get(name, None) != self.get_old(name, self.get(name)):
                 changed_properties.append(name)
 
-        if changed_properties and len(changed_properties) == 1 and changed_properties[0] == 'Options':
-            try:
-                acm.update_certificate_options(CertificateArn=self.physical_resource_id, Options=self.get('Options'))
-            except ClientError as error:
-                self.fail('{}'.format(error))
-        elif changed_properties:
-            self.fail('You can only change the "Options" of a certificate, you tried to change {}'.format(
-                ', '.join(changed_properties)))
-        else:
-            self.success('nothing to change')
+        try:
+            if 'DomainName' in changed_properties:
+                    return self.request_certificate()
+            elif changed_properties and len(changed_properties) == 1 and changed_properties[0] == 'Options':
+                try:
+                    acm.update_certificate_options(CertificateArn=self.physical_resource_id, Options=self.get('Options'))
+                except ClientError as error:
+                    self.fail('{}'.format(error))
+            elif changed_properties:
```

Comment:
why not always create a new certificate if it has changed something other than the Options?

Reply:
do we know why as such? i would like to update the stack with additional alternate names